Mathlib/Algebra/Group/Submonoid/Support.lean

```diff
@@ -0,0 +1,119 @@
+/-
+Copyright (c) 2025 Artie Khovanov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artie Khovanov
+-/
+module
+
+public import Mathlib.Algebra.Ring.Subsemiring.Defs
+public import Mathlib.RingTheory.Ideal.Defs
+
+/-!
+# Supports of submonoids
+
+Let `G` be an (additive) group, and let `M` be a submonoid of `G`.
+
+The *support* of `M` is `M ∩ -M`, the largest subgroup of `M`.
+
+When `M ∪ -M = G`, the support of `M` forms an ideal.
+
+We define supports and prove how they interact with operations.
+
+## Main definitions
+
+* `AddSubmonoid.supportAddSubgroup`: the support of a submonoid as a subgroup.
+* `AddSubmonoid.support`: the support of a submonoid as an ideal.
+
+-/
+
+@[expose] public section
+
+namespace AddSubmonoid
+
+variable {G : Type*} [AddGroup G] (M : AddSubmonoid G)
+
+end AddSubmonoid
+
+namespace Submonoid
+
+variable {G : Type*} [Group G] (M : Submonoid G)
+
+/--
+The support of a submonoid `M` of a group `G` is `M ∩ M⁻¹`, the largest subgroup contained in `M`.
+-/
+@[to_additive
+  /-- The support of a submonoid `M` of a group `G` is `M ∩ -M`,
+  the largest subgroup contained in `M`. -/]
+def supportSubgroup : Subgroup G where
+  carrier := M ∩ M⁻¹
+  one_mem' := by aesop
+  mul_mem' := by aesop
+  inv_mem' := by aesop
+
+variable {M} in
+@[to_additive]
```

Comment:
What do you think about this?
```suggestion
@[to_additive (attr := aesop simp)]
```
These kinds of case-splitting lemmas don't often work with `simp`, but they do tend to be useful for `aesop`.